askup_lti/provider.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+from django.http import Http404
+from django.shortcuts import redirect, render
+from django.urls import reverse
+from django.views.decorators.csrf import csrf_exempt
+from lti import InvalidLTIRequestError
+from lti.contrib.django import DjangoToolProvider
+from oauthlib import oauth1
+
+from askup.models import Qset
+from .models import LtiProvider, LtiUser
+from .validator import SignatureValidator
+
+log = logging.getLogger(__name__)
+
+
+@csrf_exempt
+def lti_launch(request, qset_id=None):
+    """
+    Endpoint for all requests to embed LMS content via the LTI protocol.
+
+    An LTI launch is successful if:
+    - The launch contains all the required parameters
+    - The launch data is correctly signed using a known client key/secret pair
+    """
+    try:
+        tool_provider = DjangoToolProvider.from_django_request(request=request)
+        validator = SignatureValidator()
+        ok = tool_provider.is_valid_request(validator)
+    except (oauth1.OAuth1Error, InvalidLTIRequestError, ValueError) as err:
+        ok = False
+        log.error('Error happened while LTI request: {}'.format(err.__str__()))
+    if not ok:
+        raise Http404('LTI request is not valid')
+
+    anononcement_page = render(
+        request,
+        template_name="askup_lti/announcement.html",
+        context={
+            'title': 'announcement',
+            'message': 'coming soon!',
+            'tip': 'this adaptivity sequence is about to start.',
```

Comment:
Is this `tip` going to be used anywhere? :)

Reply:
Yes. It's going to be used everywhere where launch url isn't available. 'tip' text was taken from another project and could be improved. Thanks for pointing out!